File: backend/app/schemas/room.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from app.models.user import CommunicationStyle


class RoomCreate(BaseModel):
    name: str
    is_public: bool = False


class ParticipantRead(BaseModel):
    id: uuid.UUID
    display_name: str
    communication_style: CommunicationStyle

    model_config = {"from_attributes": True}


class RoomRead(BaseModel):
    id: uuid.UUID
    name: str
    is_public: bool
    owner_id: uuid.UUID
    participants: list[ParticipantRead] = []
    created_at: datetime

    model_config = {"from_attributes": True}


class RoomUpdate(BaseModel):
    name: str | None = None
    is_public: bool | None = None

File: ios/NORMALIZER/NORMALIZER/Models/Room.swift
import Foundation

struct Room: Codable, Identifiable, Sendable {
    let id: UUID
    var name: String
    var isPublic: Bool
    let ownerId: UUID
    var participants: [Participant]
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, name, participants
        case isPublic = "is_public"
        case ownerId = "owner_id"
        case createdAt = "created_at"
    }

    struct Participant: Codable, Identifiable, Sendable {
        let id: UUID
        let displayName: String
        let communicationStyle: CommunicationStyle

        enum CodingKeys: String, CodingKey {
            case id
            case displayName = "display_name"
            case communicationStyle = "communication_style"
        }
    }
}
